src/bp/core/services/telemetry/configs.ts

```diff
@@ -0,0 +1,185 @@
+    const defaultConfigs = await this.ghostService
+      .root()
+      .readFileAsObject('/', 'botpress.config.schema.json')
+      .catch(_err => this.ghostService.root().readFileAsObject('/', 'botpress.config.schema.json'))
+
+    return this.replaceBlacklistedConfigs(botpressConfigs, defaultConfigs, botpressConfigsBlacklist)
+  }
+
+  private replaceBlacklistedConfigs(runtimeConfigs, defaultConfigs, configsBlacklist) {
+    for (const config of configsBlacklist) {
+      const defaultOrRedacted =
+        _.get(runtimeConfigs, config) == _.get(defaultConfigs, 'properties.' + config + '.default')
+          ? 'default'
+          : 'redacted'
+      _.set(runtimeConfigs, config, defaultOrRedacted)
+    }
+    return runtimeConfigs
+  }
+
+  private async getModulesConfigs(): Promise<ModuleConfigEvent[]> {
+    const bots = Array.from((await this.botService.getBots()).keys())
+    const modules = _.intersection(BUILTIN_MODULES, _.keys(process.LOADED_MODULES))
+
+    return (await Promise.map(modules, this.getConfigsByModule(bots))).reduce((acc, cur) => [...acc, ...cur])
+  }
+
+  private getConfigsByModule(bots: string[]): (module: string) => Promise<ModuleConfigEvent[]> {
+    return async module => {
+      const defaultValue = await this.moduleLoader.configReader.loadFromDefaultValues(module)
+      return await Promise.map(bots, this.getModuleConfigPerBot(module, defaultValue))
+    }
+  }
+
+  private getModuleConfigPerBot(module: string, defaultValue: any): (botId: any) => Promise<ModuleConfigEvent> {
+    return async botId => {
+      const runtimeValue = await this.moduleLoader.configReader.getForBot(module, botId)
+
+      for (const config of modulesConfigsBlacklist[module]) {
+        const defaultOrRedacted = _.get(defaultValue, config) == _.get(runtimeValue, config) ? 'default' : 'redacted'
+        _.set(runtimeValue, config, defaultOrRedacted)
+      }
+      return { botId, module, configs: runtimeValue }
+    }
+  }
+
+  private async getBotsConfigs(): Promise<BotConfigEvent[]> {
+    const defaultConfigs = await this.ghostService
+      .root()
+      .readFileAsObject('/', 'bot.config.schema.json')
+      .catch(_err => this.ghostService.root().readFileAsObject('/', 'bot.config.schema.json'))
+
+    const bots = await this.botService.getBots()
+    const configs: BotConfigEvent[] = []
+
+    for (const [id, config] of bots) {
+      const botId = calculateHash(id)
+      const botConfigs = this.formatBotConfigs(config, defaultConfigs)
+      configs.push({ botId, botConfigs })
+    }
+
+    return configs
+  }
+
+  private formatBotConfigs(configs: BotConfig, defaultConfigs) {
+    return {
+      ...this.replaceBlacklistedConfigs(configs, defaultConfigs, botConfigsBlacklist),
+      details: this.formatBotDetails(configs.details),
+      id: calculateHash(configs.id),
+      name: calculateHash(configs.name)
+    }
+  }
+
+  private formatBotDetails(details: BotDetails) {
+    if (_.isEmpty(details)) {
```

Comment:
To avoid repeating yourself
````js
const details = ['website', 'phoneNumber', 'termsConditions']
return details.reduce((acc, curr, {...acc, [curr]: 'default'})
```

Reply:
fixed in: eaf1494a28fc2d5b7f24e0bc874148bc92a38798